plugins/curl/address.ml

```diff
@@ -0,0 +1,40 @@
+open Current.Syntax
+
+[@@@ocaml.warning "-3"]
+
+type t = {
+  fqdn : string;
+  ip : string;
+} [@@deriving yojson]
+
+[@@@ocaml.warning "+3"]
+
+let equal = (=)
+let compare = compare
+let hash t = t.fqdn ^ "=" ^ t.ip
+let digest { fqdn; ip } = Fmt.str "%s=%s" fqdn ip
+
+let pp f { fqdn = _; ip } = Fmt.pf f "%s" ip
+
+let pp_short f t =
+  Fmt.string f @@ Astring.String.with_range ~len:6 (hash t)
+
+let expand a =
+  Current.component "list" |>
+  let> a = a in
+  Current.Primitive.const (
+    String.split_on_char '\n' a.Dig.Value.result |>
```

Comment:
```suggestion
let expand dns  =
  Current.component "list" |>
  let> dns = dns in
  Current.Primitive.const (
    String.split_on_char '\n' dns.Dig.Value.result |>
```
Proposition to make it more readable. Feel free to ignore :+1:

Reply:
I went with `a` for address but happy to change it to `dns`